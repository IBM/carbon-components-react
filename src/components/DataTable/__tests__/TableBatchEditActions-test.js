import React from 'react';
import { shallow } from 'enzyme';
import InlineLoading from '../../InlineLoading';
import { TableBatchEditActions } from '../';

describe('DataTable.TableBatchEditActions', () => {
  const onSave = jest.fn();
  const onCancel = jest.fn();

  it('should render', () => {
    const wrapper = shallow(
      <TableBatchEditActions onSave={onSave} onCancel={onCancel} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render custom attributes on the top-level node', () => {
    const wrapper = shallow(
      <TableBatchEditActions
        data-foo="Foo"
        onSave={onSave}
        onCancel={onCancel}
      />
    );
    expect(wrapper.prop('data-foo')).toBe('Foo');
  });

  it('should show the action bar when activated', () => {
    const wrapper = shallow(
      <TableBatchEditActions
        shouldShowBatchActions
        onSave={onSave}
        onCancel={onCancel}
      />
    );
    expect(wrapper.hasClass('bx--batch-actions--active')).toBe(true);
  });

  it('should show the spinner during saving', () => {
    const wrapper = shallow(
      <TableBatchEditActions
        shouldShowBatchActions
        saving
        onSave={onSave}
        onCancel={onCancel}
      />
    );
    expect(
      wrapper
        .find(InlineLoading)
        .matchesElement(
          <InlineLoading description="Saving..." success={false} />
        )
    ).toBe(true);
  });

  it('should show the checkmark once saving completes', () => {
    const wrapper = shallow(
      <TableBatchEditActions
        shouldShowBatchActions
        saving
        saved
        onSave={onSave}
        onCancel={onCancel}
      />
    );
    expect(
      wrapper
        .find(InlineLoading)
        .matchesElement(<InlineLoading description="Saving..." success />)
    ).toBe(true);
  });
});
