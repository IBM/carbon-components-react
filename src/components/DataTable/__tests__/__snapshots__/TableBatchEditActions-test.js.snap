// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableBatchEditActions should render 1`] = `
<div
  className="bx--batch-actions"
>
  <div
    className="bx--batch-edit-status"
  >
    <p
      className="bx--batch-summary__para"
    >
      Editing...
    </p>
  </div>
  <div
    className="bx--batch-summary bx--batch-edit-summary"
  >
    <TableBatchAction
      className="bx--batch-edit-summary__save"
      icon={
        Object {
          "height": "16",
          "id": "icon--save",
          "name": "icon--save",
          "styles": "",
          "svgData": Object {
            "circles": "",
            "ellipses": "",
            "paths": Array [
              Object {
                "d": "M12 1.597v2.406a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V1H1v14h3V9a1 1 0 0 1 1-1h6a1 1 0 0 1 1 1v6h3V4.702l-3-3.105zM1 0h10.848L16 4.298V15a1 1 0 0 1-1 1H1a1 1 0 0 1-1-1V1a1 1 0 0 1 1-1zm10 9H5v5.992l6 .007V9zM5 1v3.003h6V1H5z",
                "fill-rule": "nonzero",
              },
            ],
            "polygons": "",
            "polylines": "",
            "rects": "",
          },
          "tags": "icon--save",
          "viewBox": "0 0 16 16",
          "width": "16",
        }
      }
      iconDescription="Add"
      kind="ghost"
      onClick={[MockFunction]}
      small={true}
    >
      Save
    </TableBatchAction>
    <button
      className="bx--batch-summary__cancel bx--batch-edit-summary__cancel"
      onClick={[MockFunction]}
    >
      Cancel
    </button>
  </div>
</div>
`;
